app/javascript/app/components/anchor-nav/anchor-nav-component.jsx

```diff
@@ -22,56 +22,57 @@ const AnchorNav = props => {
+                className: theme.link,
+                activeClassName: theme.linkActive,
+                to: {
+                  search: link.search || query,
+                  pathname: link.path,
+                  hash: link.hash
+                }
+              };
+              if (link.activeQuery) {
+                linkProps.isActive = (match, location) => {
+                  const activeSearchQuery = qs.parse(location.search)[
+                    link.activeQuery.key
+                  ];
+                  const linkSearchQuery = link.activeQuery.value;
+                  return (
+                    activeSearchQuery === linkSearchQuery ||
+                    (index === 0 && !activeSearchQuery)
+                  );
+                };
               }
-            };
-            if (link.activeQuery) {
-              linkProps.isActive = (match, location) => {
-                const activeSearchQuery = qs.parse(location.search)[
-                  link.activeQuery.key
-                ];
-                const linkSearchQuery = link.activeQuery.value;
+              if (useRoutes) {
+                linkProps.exact = true;
                 return (
-                  activeSearchQuery === linkSearchQuery ||
-                  (index === 0 && !activeSearchQuery)
+                  <NavLink {...linkProps} replace>
+                    {link.label}
+                  </NavLink>
                 );
-              };
-            }
-            if (useRoutes) {
-              linkProps.exact = true;
+              }
+              linkProps.isActive = (match, location) =>
+                link.hash === activeSection ||
+                (link.defaultActiveHash &&
+                  (`#${link.hash}` === location.hash ||
+                    (!location.hash && index === 0)));
               return (
-                <NavLink {...linkProps} replace>
+                <NavHashLink
+                  {...linkProps}
+                  smooth
+                  scroll={el => {
+                    el.scrollIntoView(true);
+                    if (offset) window.scrollBy(0, offset[index]);
+                  }}
+                  replace
+                >
                   {link.label}
-                </NavLink>
+                </NavHashLink>
               );
-            }
-            linkProps.isActive = (match, location) =>
-              link.hash === activeSection ||
-              (link.defaultActiveHash &&
-                (`#${link.hash}` === location.hash ||
-                  (!location.hash && index === 0)));
-            return (
-              <NavHashLink
-                {...linkProps}
-                smooth
-                scroll={el => {
-                  el.scrollIntoView(true);
-                  if (offset) window.scrollBy(0, offset[index]);
-                }}
-                replace
-              >
-                {link.label}
-              </NavHashLink>
-            );
-          })}
+            })}
```

Comment:
Hey @Bluesmile82, could you move all this logic to the container component? Too much stuff is going on here and it's harder to read I think.

Reply:
I just added the links clause because it was breaking on some edge cases. Let's add a task to clean this if needed

Reply:
I'll add the task then, otherwise we will postpone it on never :)